File: le-transform/src/main/java/com/latticeengines/transform/v2_0_25/functions/StdVisidbDsStateIsInPlains.java
package com.latticeengines.transform.v2_0_25.functions;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Map;

import org.apache.commons.lang3.StringUtils;

import com.latticeengines.transform.exposed.RealTimeTransform;
import com.latticeengines.transform.exposed.metadata.ApprovedUsage;
import com.latticeengines.transform.exposed.metadata.Category;
import com.latticeengines.transform.exposed.metadata.FundamentalType;
import com.latticeengines.transform.exposed.metadata.StatisticalType;
import com.latticeengines.transform.exposed.metadata.Tag;
import com.latticeengines.transform.exposed.metadata.TransformMetadata;

public class StdVisidbDsStateIsInPlains implements RealTimeTransform {

    private static final long serialVersionUID = 1147678318164791088L;
    static HashSet<String> valueMap = new HashSet<String>(Arrays.asList("MO", "MN", "ND", "NE", "KS", "IA", "SD"));

    public StdVisidbDsStateIsInPlains() {
        
    }
    
    public StdVisidbDsStateIsInPlains(String modelPath) {
    }

    public Object transform(Map<String, Object> arguments, Map<String, Object> record) {
        String column = (String) arguments.get("column");
        Object n = record.get(column);

        if (n == null)
            return false;

        String s = n.toString().trim().toUpperCase();

        return getDSStateIsInPlains(s);
    }

    public static Boolean getDSStateIsInPlains(String state) {
        if (StringUtils.isEmpty(state))
            return false;

        return valueMap.contains(state);
    }

    @Override
    public TransformMetadata getMetadata() {
        TransformMetadata metadata = new TransformMetadata();
        metadata.setApprovedUsage(ApprovedUsage.MODEL_ALLINSIGHTS);
        metadata.setCategory(Category.LEAD_INFORMATION);
        metadata.setFundamentalType(FundamentalType.BOOLEAN);
        metadata.setStatisticalType(StatisticalType.NOMINAL);
        metadata.setDescription("Region: Plains");
        metadata.setDisplayName("Region: Plains");
        metadata.setTags(Tag.INTERNAL_TRANSFORM);
        return metadata;
    }
}

File: le-serviceflows/core/src/main/resources/com/latticeengines/serviceflows/core/transforms/std_visidb_ds_state_isInPlains.py
#This function looks for words and returns TRUE if there is a word map, FALSE otherwise

def metadata():
    return {  'ApprovedUsage'   : 'ModelAndAllInsights'
            , 'Category'        : 'Lead Information'
            , 'FundamentalType' : 'boolean'
            , 'Description'     : 'Region: Plains'
            , 'DisplayName'     : 'Region: Plains'            
            , 'StatisticalType' : 'nominal'
            , 'Tags'            : 'Internal' }

#simple test for membership
def std_visidb_ds_state_isInPlains(x):
    return x in set(['MO', 'MN', 'ND', 'NE', 'KS', 'IA', 'SD'])

#function to call
def transform(args, record):
    column = args["column"]
    value = record[column]
    return std_visidb_ds_state_isInPlains(value)
